scripts/bootloader/hash.py

```diff
@@ -8,22 +8,29 @@
 import hashlib
 import sys
 import argparse
+from intelhex import IntelHex
 
 
 def parse_args():
     parser = argparse.ArgumentParser(
-        description="Hash data from stdin or file.",
+        description="Hash data from file.",
         formatter_class=argparse.RawDescriptionHelpFormatter)
 
-    parser.add_argument("--infile", "-i", "--in", "-in", required=False,
-                        type=argparse.FileType('rb'), default=sys.stdin.buffer,
-                        help="Hash the contents of the specified file instead of stdin.")
-
-    args = parser.parse_args()
-
-    return args
+    parser.add_argument("--infile", "-i", "--in", "-in", required=True,
+                        help="Hash the contents of the specified file."
```

Comment:
I think this could be made clearer. Something like:

"Hash the contents of the specified file. If a *.hex file is given, the contents will first be converted to binary. For all other file types, no conversion is done."